src/content/developers/docs/smart-contracts/upgrading-smart-contracts/index.md

```diff
@@ -0,0 +1,159 @@
+
+3. Proxy patterns adopt a catch-all approach to access controls. An entity with access to upgrade functions can change the _entire_ contract. But the diamond pattern enables a modular permissions approach, where you can restrict entities to upgrading certain functions within a smart contract. 
+
+[More on the diamond pattern](https://eip2535diamonds.substack.com/p/introduction-to-the-diamond-standard?s=w). 
+
+## Pros and cons of upgrading smart contracts {#pros-and-cons-of-ugrading-smart-contracts}
+
+| Pros                                | Cons                                                                             |
+| ----------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------ |
+| A smart contract upgrade can make it easier to fix vulnerabilities discovered in the post-deployment phase.            | Upgrading smart contracts negates the idea of code immutability, which has implications for decentralization and security.                                          |
+| Developers can use upgrades to add new features to smart contract | Users must trust developers not to modify smart contracts arbitrarily.  |
+| Smart contract upgrades can improve safety for end-users since bugs can be fixed quickly.                              | Programming upgrade functionality into smart contracts adds another layer of complexity and increases the possibility of critical flaws.                        |
+| The opportunity to upgrade smart contracts may encourage developers to launch projects faster without doing due dilligence during the development phase. | Insecure access control or centralization in smart contracts can make it easier for malicious actors to perform unauthorized upgrades.
+
+## Considerations for upgrading smart contracts {#considerations-for-upgrading-smart-contracts}
+
+1. Use secure access control/authorization mechanisms to prevent unauthorized smart contract upgrades, especially if using the proxy pattern, strategy pattern, or data separation.
+
+2. Upgrading smart contracts is a complex activity and requires a high level of dilligence to prevent the introduction of vulnerabilities. 
+
+3. Reduce trust assumptions by decentralizing the process of implementing upgrades. Possible strategies include using a [multi-sig wallet contract](/developers/docs/smart-contracts/#multisig) to control upgrades, or requiring [members of a DAO](/dao/) to vote on approving the upgrade. 
+
+4. Be aware of the costs involved in upgrading contracts. For instance, copying state (e.g., user balances) from an old contract to a new contract during contract migration may require more than one transaction, meaning more gas fees. 
+
+## Resources {#resources}
+**OpenZeppelin Upgrades OS - _A suite of tools for deploying and securing upgradeable smart contracts._**
```

Comment:
It's called **OpenZeppelin Upgrades plugins** now.

Reply:
Fixed this now.